public/js/index.js

```diff
@@ -2120,3 +2120,55 @@ function initNavbarContentToggle() {
         }
     });
 }
+
+function deleteDependencyModal(id, type) {
+    $('.remove-dependency')
+        .modal({
+            closable: false,
+            duration: 200,
+            onApprove: function () {
+                $('#removeDependencyID').val(id);
+                $('#dependencyType').val(type);
+                $('#removeDependencyForm').submit();
+            }
+        }).modal('show')
+    ;
+}
+
+function showAddDependencyModal() {
+    // Show the modal
+    $('.add-dependency')
+        .modal({
+            duration: 200,
+            onApprove: function() {
+                $('#addDependencyForm').submit();
+
+            }
+        }).modal('show')
+    ;
+
+    buildIssuesList();
+}
+
+function buildIssuesList() {
+    // Get a list of issues
+    var repolink = $('#repolink').val();
+    var issueIndex = $('#issueIndex').val();
+
+    if(repolink !== undefined) {
+        $.getJSON('/api/v1/repos' + repolink + '/issues', function (data) {
+
+            $.each(data, function (i, issue) {
+                if (issue.number != issueIndex) {
+                    $('.new-dependency-dropdown').append('<div class="item" data-value="' + issue.id + '"><b>#' + issue.number + '</b> ' + issue.title + '</div>');
+                }
+            });
+
+            $('.new-dependency-drop-list').dropdown({
+                fullTextSearch: true
+            });
+        });
+    }
+}
+
+buildIssuesList();
```

Comment:
Can't we call this lazily? (only when one clicks on the dropdown)

Reply:
Sure. 

I personally see this more like a temporary solution until we can search for the issues via api, which is why I'm loading the issues via javascript instead of loading them directly inside the template.